src/cryptonote_core/blockchain.cpp

```diff
@@ -393,18 +393,7 @@ bool Blockchain::init(BlockchainDB* db, const network_type nettype, bool offline
     MINFO("Blockchain not loaded, generating genesis block.");
     block bl = boost::value_initialized<block>();
     block_verification_context bvc = boost::value_initialized<block_verification_context>();
-    if (m_nettype == TESTNET)
-    {
-      generate_genesis_block(bl, config::testnet::GENESIS_TX, config::testnet::GENESIS_NONCE);
-    }
-    else if (m_nettype == STAGENET)
-    {
-      generate_genesis_block(bl, config::stagenet::GENESIS_TX, config::stagenet::GENESIS_NONCE);
-    }
-    else
-    {
-      generate_genesis_block(bl, config::GENESIS_TX, config::GENESIS_NONCE);
-    }
+    generate_genesis_block(bl, config_table[m_nettype].GENESIS_TX, config_table[m_nettype].GENESIS_NONCE);
```

Comment:
You will need to ensure each of those paths can't dereference config_table[FAKECHAIN] (ie, core when in tests), unless you've done so already.

Reply:
OK, I made it such that `config_table[MAINNET]` is the same as `config_table[FAKECHAIN]`. Does it look fine?

Reply:
I don't know whether this kind of self reference is standard or not.

Reply:
I confirmed all the core tests pass.